src/mlpack/core.hpp

```diff
@@ -120,4 +120,12 @@
   #define MLPACK_USE_CX11
 #endif
 
+#if !defined(MLPACK_COUT_STREAM)
```

Comment:
Should I replace all the `std::cout` (used for generating bindings) with `MLPACK_COUT_STREAM`?

Reply:
Yes.  Otherwise we cannot go to CRAN.

Reply:
Sure I will replace. But I think so std::cout in bindings folder is only used for writing bindings only, and most of these might not even touch by R bindings. But for uniformity we must replace all of these. 
Thank You.

Reply:
It's not some obscure rule:  _If_ it ends up in the object code of the package loaded by R, it _will_ be detected and _should_ therefore be replaced.

Test via `R CMD check nameOfYourTarball.gz`.

Reply:
Yes, In ctest I am testing the bindings with `R CMD check` only. As internally `ctest -R r_binding_test --verbose` invokes R CMD check only.

Reply:
@rcurtin yes. Moreover,  maybe check / use `ARMA_COUT_STREAM` or `ARMA_CERR_STREAM` if set?

http://arma.sourceforge.net/docs.html#logging

Reply:
Sure, that seems fine to me too; I just want to make sure we document the functionality somewhere in case anyone else ever needs it. :smile:

Reply:
I have added documentation here. One more thing I shifted this in `prereqs.hpp`. replaced MLPACK_COUT_STREAM to every file of mlpack. I will take a look once more for confirmation.

Reply:
Actually I think it shouldn't be necessary to replace any of the `std::cout` usage in `src/mlpack/bindings/go/`, `src/mlpack/bindings/julia/`, `src/mlpack/bindings/cli/`, `src/mlpack/bindings/python/`, or `src/mlpack/bindings/markdown/`.  The reason I say that is that those files aren't really a part of mlpack and they won't be distributed with the R package; those are used in specially-compiled utilities that need to print to stdout to generate the binding files.  So no user would ever reasonably want to use anything other than `std::cout` there, and it won't trigger the R warnings since it won't be a part of what we package for CRAN.  Would you mind reverting those changes?  (Or let me know if I overlooked something. :+1:)

Reply:
If they are not in the shared library the R package loads then we're good.

Reply:
Reverted the changes in all the mentioned folders.